Core/include/Acts/Geometry/GridPortalLink.hpp

```diff
@@ -0,0 +1,614 @@
+  /// @param direction The merging direction
+  /// @param logger The logger to use for messages
+  /// @return The merged grid portal link or nullptr if grid merging did not succeed
+  /// @note The returned pointer can be nullptr, if the grids
+  ///       given are not mergeable, because their binnings are
+  ///       not compatible. This is not an error case, and needs
+  ///       to be handled by th caller! Invalid input is handled
+  ///       via exceptions.
+  static std::unique_ptr<PortalLinkBase> merge(
+      const GridPortalLink& a, const GridPortalLink& b, BinningValue direction,
+      const Logger& logger = getDummyLogger());
+
+  /// Return the associated grid in a type-erased form
+  /// @return The grid
+  virtual const IGrid& grid() const = 0;
+
+  /// Set the volume on all grid bins
+  /// @param volume The volume to set
+  virtual void setVolume(TrackingVolume* volume) = 0;
+
+  /// Get the number of dimensions of the grid
+  /// @return The number of dimensions
+  virtual unsigned int dim() const = 0;
+
+  /// Expand a 1D grid to a 2D one, by using the provided axis along the
+  /// *missing* direction.
+  /// @param other The axis to use for the missing direction
+  /// @return A unique pointer to the 2D grid portal link
+  virtual std::unique_ptr<GridPortalLink> make2DGrid(
+      const IAxis* other) const = 0;
+
+  /// The binning direction of the grid
+  /// @note For 2D grids, this will always be the loc0
+  ///       direction, depending on the surface type.
+  /// @return The binning direction
+  BinningValue direction() const { return m_direction; }
+
+  /// Helper function to fill the bin contents after merging.
+  /// This called by the merging routine, and requires access to the internal
+  /// grid state.
+  /// @param a The first grid portal link
+  /// @param b The second grid portal link
+  /// @param merged The merged grid portal link
+  /// @param direction The merging direction
+  /// @param logger The logger to use for messages
+  static void fillMergedGrid(const GridPortalLink& a, const GridPortalLink& b,
+                             GridPortalLink& merged, BinningValue direction,
+                             const Logger& logger);
+
+  /// Helper function that prints a textual representation of the grid with the
+  /// volume names.
+  /// @param os The output stream
+  void printContents(std::ostream& os) const;
+
+ protected:
+  /// Helper function to check consistency for grid on a cylinder surface
+  /// @param cyl The cylinder surface
+  void checkConsistency(const CylinderSurface& cyl) const;
+
+  /// Helper function to check consistency for grid on a disc surface
+  /// @param disc The disc surface
+  void checkConsistency(const DiscSurface& disc) const;
+
+  /// Expand a 1D grid to a 2D one for a cylinder surface
+  /// @param surface The cylinder surface
+  /// @param other The axis to use for the missing direction,
+  ///              can be null for auto determination
```

Comment:
It could be, but this is an internal interface for the grid merging, and I though explicit arguments at the call site make it easier to follow what's going on.